presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergFileWriterFactory.java

```diff
@@ -165,9 +165,9 @@ private IcebergFileWriter createOrcWriter(
     {
         try {
             FileSystem fileSystem = hdfsEnvironment.getFileSystem(session.getUser(), outputPath, jobConf);
-            OrcDataSink orcDataSink = new OutputStreamOrcDataSink(fileSystem.create(outputPath));
+            OrcDataSink orcDataSink = hdfsEnvironment.doAs(session.getUser(), () -> new OutputStreamOrcDataSink(fileSystem.create(outputPath)));
             Callable<Void> rollbackAction = () -> {
-                fileSystem.delete(outputPath, false);
+                hdfsEnvironment.doAs(session.getUser(), () -> fileSystem.delete(outputPath, false));
```

Comment:
should we also add `recoverAction` to `IcebergRecordFileWriter`  [created-at](https://github.com/prestosql/presto/blob/master/presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergFileWriterFactory.java#L147-L157 ) 
as otherwise.

[this] (https://github.com/prestosql/presto/blob/master/presto-hive/src/main/java/io/prestosql/plugin/hive/RecordFileWriter.java#L197) might fail as its running with `Authentication` ( doAs )

Reply:
Sorry for commenting bit late.

Reply:
2nd comment is:

the same above problem can occur at this point as well in `RecordFileWriter`: [code](https://github.com/prestosql/presto/blob/master/presto-hive/src/main/java/io/prestosql/plugin/hive/RecordFileWriter.java#L130)

Flow to reach this methods are from [HiveWriter1] (https://github.com/prestosql/presto/blob/84e2883886edeb5a34b89b2136b521127486ee96/presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriter.java#L62),  and [IcebergPageSink](https://github.com/prestosql/presto/blob/84e2883886edeb5a34b89b2136b521127486ee96/presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergPageSink.java#L179)

Reply:
Hmm makes sense. We can add a `rollbackAction` in `RecordFileWriter` like what we have in `OrcFileWriter`. It actually depends on whether we should wrap deletes in `doAs`. I feel we can do it in a separate PR.

Reply:
Sure, though but we are wrapping deletes in `doAs`, thats why I pointed.
also in the second comment, ` path.getFileSystem(conf).getFileStatus(path).getLen();` is called without `doAs`. So, that can also be a problem. WDYT ?

Reply:
If the above case is not true, and we don't need to use `doAs`.
Then its LG for me, and thanks for working on this !!

Reply:
@manishmalhotrawork Thanks for letting us know. I think we can do the fix in a separate PR as it involves changes in `presto-hive`, so we're good for merging this one as of now.

Reply:
@lxynov thanks.
but if we see: https://github.com/prestosql/presto/pull/4460#discussion_r456142868
that flow is being called from `IcebergSink` as well. 
So, iceberg writes can still create problem.

Reply:
@electrum can you please see my comment as well, and if it's not a problem( or can be taken care in future).
then can we please merge this fix.

Reply:
We can follow up if it's a problem.

Reply:
@electrum thanks !